scylla-cdc-driver3/src/main/java/com/scylladb/cdc/cql/driver3/Driver3SchemaBuilder.java

```diff
@@ -12,125 +13,477 @@
+         *             was renamed. That means that all other columns are not
+         *             changed. If the queried CDC column is a part of partition key
+         *             of base schema, that means that it was a partition
+         *             key (as of CDC schema time) - partition columns
+         *             were not modified (clustering column was modified).
+         *             Similarly for regular columns. But, if the column
+         *             is missing from the base table schema, that means
+         *             that we are seeing a old clustering column.
+         *             </td>
+         *     </tr>
+         *     <tr>
+         *         <td>Base schema newer than CDC log schema</td>
+         *         <td>Column data type changed</td>
+         *         <td>The column data type change does not change
+         *             the column set (column names, whether the columns
+         *             are in the primary key). Therefore, the schemas
+         *             are consistent with regard to column type.</td>
+         *     </tr>
+         *     <tr>
+         *         <td>CDC log schema newer than base schema</td>
+         *         <td>Column added</td>
+         *         <td>Only regular columns can be added. This method
+         *             will not find this added column in the
+         *             base schema primary key, so it will
+         *             correctly return REGULAR.</td>
+         *     </tr>
+         *     <tr>
+         *         <td>CDC log schema newer than base schema</td>
+         *         <td>Column removed</td>
+         *         <td>This method will never be called for that removed column,
+         *             as the column name was not present in the CDC log schema (caller).</td>
+         *     </tr>
+         *     <tr>
+         *         <td>CDC log schema newer than base schema</td>
+         *         <td>Column renamed</td>
+         *         <td>The same logic as in "Column renamed" when
+         *             the base schema is newer than CDC log schema.
+         *             The reasoning is a bit symmetrical
+         *             (ck_new in CDC log schema rather than
+         *             base table schema).</td>
+         *     </tr>
+         *     <tr>
+         *         <td>CDC log schema newer than base schema</td>
+         *         <td>Column data type changed</td>
+         *         <td>The same logic as in "Column data type changed" when
+         *             the base schema is newer than CDC log schema.</td>
+         *     </tr>
+         * </table>
+         *
+         * @param columnName the column name to get the column type for.
+         * @param cdcColumnDefinitions the column definitions for the CDC log table.
+         * @return the column type in the base table.
+         */
+        public ChangeSchema.ColumnType getColumnType(String columnName, ColumnDefinitions cdcColumnDefinitions) {
```

Comment:
Could you add a new commit before this one that changes `ColumnType` to `ColumnKind` (and then this function would be called getColumnKind)? Using ColumnType is reaaally confusing.

Reply:
Even better: getBaseColumnKind?

Reply:
In the amended commit changed the naming to ColumnKind in the entire project (and all comments, etc.). This method is now called `getColumnKind`. I did not find it necessary to name it `getBaseColumnKind`, because a class it is in (`BaseTableSchema`) already implies this.